crates/diesel_models/src/authorization.rs

```diff
@@ -57,6 +57,20 @@ pub struct AuthorizationUpdateInternal {
     pub connector_authorization_id: Option<String>,
 }
 
+impl AuthorizationUpdateInternal {
+    pub fn create_authorization(self, source: Authorization) -> Authorization {
+        Authorization {
+            status: self.status.unwrap_or(source.status),
+            error_code: self.error_code.unwrap_or(source.error_code),
+            error_message: self.error_message.unwrap_or(source.error_message),
```

Comment:
similarly for error_message and modified_at, we should use `or` instead of unwrap_or because they expect option values